Sally7/S7Connection.cs

```diff
@@ -98,12 +100,31 @@ public void Dispose()
         public async Task OpenAsync()
         {
             await TcpClient.ConnectAsync(host, IsoOverTcpPort).ConfigureAwait(false);
+            await openAsync();
+        }
+
+        public async Task OpenAsync(double connectionTimeout)
```

Comment:
I would recommend using TimeSpan for the timeout. Otherwise I don't know what unit you expect here.